include/swift/AST/DiagnosticsSema.def

```diff
@@ -330,7 +330,7 @@ ERROR(cannot_convert_initializer_value_nil,none,
       "'nil' cannot initialize specified type %0", (Type))
 
 ERROR(cannot_convert_to_return_type,none,
-      "cannot convert return expression of type %0 to return type %1",
+      "return expressions for types %0 and %1 are not interconvertible",
```

Comment:
IIUC the feedback from @xedin, the discussion about rewording was not about only re-phrasing but add more info about the types involved.

Reply:
Indeed, notes are to be added, as I said above, I have only added in the test file at the moment. Will add it in the code as well to reflect the same.

Reply:
@LucianoPAlmeida 
For this particular case:

```Swift
struct S {
  var str: String
}

func test(s: S?) {
  let _: Int? = s?.str

  func f(_: Int?) {}
  f(s?.str)
}
```

I don't get the expected `error`, it still gives an `ambiguous` diagnostic message:

```
error: type of expression is ambiguous without a type annotation
```

Reply:
Seems like fixes are being correctly added in repairFailures but something is happening in the middle resulting in not a solution formed with those fixes.

Reply:
Yeah, in `diagnoseFailureFor`, it's falling through this call:

```C++
DE.diagnose(expr->getLoc(), diag::type_of_expression_is_ambiguous)
        .highlight(expr->getSourceRange());
```

Reply:
Actually I added breakpoints in `repairFailures `, it actually never goes through `path.back().is<LocatorPathElt::ContextualType>()` as a successful check. Hence, it does through the ambiguous diagnosis.